### PR TITLE
Profile agreements

### PR DESCRIPTION
**Pull request by huwshimi**
Add the list of agreed to terms on the profile page.

### CODE REVIEW

**Comment by huwshimi** [comment]
Requires https://github.com/juju/juju-gui/pull/1802 to land to QA correctly.

The links to terms will be added as a followup.

QA:
- connect to GUIMAAS
- agree to some terms (e.g. `app.terms.addAgreement([{name: 'lorem-ipsum', revision: 4}], (error, data) => {console.log(error, data)})`)
- open your profile
- you should see the terms you have agreed to

**Comment by jujugui** [comment]
Test PASSed.
Refer to this link for build results (access rights to CI server needed): 
http://ci.jujugui.org:8080//job/juju-gui/5319/
Test PASSed.

File: jujugui/static/gui/src/app/components/expanding-row/_expanding-row.scss [hunk 1]
```diff
@@ -1,4 +1,6 @@
 .expanding-row {
+    cursor: pointer;
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/expanding-row/_expanding-row.scss]: Thanks

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 2]
```diff
@@ -585,6 +622,31 @@ YUI.add('user-profile', function() {
...
+      @returns {Array} The markup for the row.
+    */
+    _generateAgreementRow: function(agreement) {
+      var term = agreement.term;
+      var createdAt = agreement.createdAt;
+      var date = createdAt.getUTCFullYear() + '-' +
+        ('0' + (createdAt.getUTCMonth() + 1)).slice(-2) + '-' +
+        ('0' + createdAt.getUTCDate()).slice(-2);
+      return (
+        <li className="user-profile__list-row twelve-col"
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Don't think this requires a `twelve-col` as it is inside a ul with `twelve-col`

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js [hunk 3]
```diff
@@ -606,6 +668,24 @@ YUI.add('user-profile', function() {
...
 
     /**
+      Generate the header for the agreements.
+
+      @method _generateAgreementHeader
+      @returns {Array} The markup for the header.
+    */
+    _generateAgreementHeader: function() {
+      return (
+        <li className="user-profile__list-header twelve-col">
```
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Same here

**Comment by anthonydillon** [comment]
Thanks Huw. :+1: with a comment about grid overuse. Preferably we want to keep the grid to a single level and when required only nest by one level.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 2 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: I found that if it wasn't a twelve-col the element would collapse, so the borders ended up stacked at the top.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 3 above)
Comment (huwshimi) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: As above.

File: jujugui/static/gui/src/app/components/user-profile/user-profile.js (hunk 2 above)
Comment (anthonydillon) [inline jujugui/static/gui/src/app/components/user-profile/user-profile.js]: Ok, sounds ok then.

**Comment by hatched** [comment]
Looks good, thanks @huwshimi :+1: 
:shipit:

**Comment by jujugui** [comment]
Status: merge request accepted. Url: http://ci.jujugui.org:8080/job/juju-gui-merge

**Pull request merged**